scRNA-seq-advanced/setup/integration-setup.Rmd

```diff
@@ -0,0 +1,402 @@
+#harmony::HarmonyMatrix(
+#  data_mat  = logcounts(combined_sce), 
+#  meta_data = colData(combined_sce)$batch
+#)
+
+# The result is just a bunch of PCs!
+head(harmony_result)
+
+# Let's store this directly into our `combined_sce` object, making sure to name it well!
+reducedDim(combined_sce, "harmony_PCA") <- harmony_result
+
+# Calculate UMAP with the right name for PCs:
+# We do NOT want to overwrite the uncorrected UMAP! Make sure to name this differently with `name`
+combined_sce <- scater::runUMAP(combined_sce, dimred = "harmony_PCA", name = "harmony_UMAP")
+```
+
+#### The before/after reveal, again!
+
+
+```{r}
+# Before
+scater::plotReducedDim(combined_sce,
+                       dimred = "UMAP",
+                       colour_by = "batch",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.4) +
+  ggplot2::ggtitle("Before integration")
+
+
+# After
+scater::plotReducedDim(combined_sce,
+                       dimred = "harmony_UMAP",
+                       colour_by = "batch",
+                       # Some styling to help us see the points:
+                       point_size = 0.5,
+                       point_alpha = 0.4) +
+  ggplot2::ggtitle("After integration with harmony")
+
+```
+
+
+## Comparing back to celltypes
+
+In an ideal world, we'd have cell types to refer to in UMAPs as well; well-integrated datasets should show more clustering of cell types and less clustering of batches.
+Surprise, it's an ideal world!
+
+
+Before we look at cell types, it might be nice to have all our UMAPs in one place.
+Since we already have the unintegrated and harmony UMAPs in `combined_sce`, let's just go ahead and add `fastmnn` UMAPs into that SCE as well.
+We can use `combined_sce` going forward.
+```{r}
+reducedDim(combined_sce, "fastmnn_UMAP") <- reducedDim(integrated_sce, "UMAP")
+```
+
+
+Magic celltypes!
+Note that these subtypes only have the current libraries in them but this super quick to update with, for example, exercise libraries.
+Also note, we could go a bit rogue and start with SCEs that have the celltypes in `colData` already?
+```{r}
+celltypes <- readr::read_tsv(file.path(data_dir, "celltypes.tsv"))
+celltypes
+```
+
+We'd like to add these celltypes into the `colData` slot of `combined_sce` so that we can color UMAPs by it.
+Earlier we formatted the `colData` rownames as `batch-barcode`, and that information is also in the celltypes TSV.
+
+```{r}
+# Let's wrangle celltypes a bit to have exactly matching information:
+celltypes <- celltypes %>%
+  dplyr::mutate(cell_id = glue::glue("{library}-{barcode}")) %>%
+  dplyr::select(cell_id, celltype)
+celltypes
+
+# Let's add it into the colData with some wrangling to join
+# code suggestions welcome here!
+colData(combined_sce) <- as.data.frame(colData(combined_sce)) %>%
+  tibble::rownames_to_column("cell_id") %>%
+  dplyr::inner_join(celltypes) %>%
```

Comment:
In case there are any cell types missing, we want a left join. Probably also worth checking dimensions before adding back, just to be safe.

```suggestion
  dplyr::left_join(celltypes) %>%
```

Alternatively, maybe something like:
```
celltypes <- filter(celltypes, cell_id %in% colnames(combined_sce))
colData(combined_sce)$celltype <- NA
colData(combined_sce)[celltypes$cell_id, "celltype"] <- celltypes$celltype
```
This is untested... and I like the dplyr version better. Also, this is likely to end up in the setup if we follow my earlier thoughts...